guide/14-deep-learning/how_pspnet_works_and_its_implementation_details_within_arcgis.ipynb

```diff
@@ -0,0 +1,234 @@
+{
```

Comment:
<p>My review is based on an existing guide at https://developers.arcgis.com/python/guide/how-unet-works/</p><p> </p><p>I would write a title that says:</p><p> </p><p> </p><p> </p><p># How PSPNet works?</p><p> </p><p> </p><p>## Introduction (subheading1)</p><p> </p><p>In the guide [How u-net works](https://developers.arcgis.com/python/guide/how-unet-works/), we have learned in detail about semantic segmentation. There are many semantic segmentation algorithms such as U-net, PSPNet, Feature Pyramid Network (FPN), etc. In this guide, we will mainly focus on Pyramid scene parsing network (PSPNet) [1] which is one of the most well-recognized image segmentation algorithms and many of the ideas are shared among other algorithms.</p><p> </p><p>Note: To follow the guide below, we assume that you have some basic understanding of the convolutional neural networks (CNN) concept. You can refresh your CNN knowledge by going through this short paper “[A guide to convolution arithmetic for deep learning](https://arxiv.org/pdf/1603.07285.pdf)".</p><p> </p><p>## Why is PSPNet better than FCN? (subheading2)</p><p> </p><blockquote>It resolves the problem of parsing scenes based on semantic segmentation. It takes into account the global context of the image to predict the local level predictions hence gives better performance on benchmark datasets like PASCAL VOC 2012 and cityscapes. The model was needed because FCN based pixel classifiers were not able to capture the context of the whole image.</blockquote><p>Comment: I think we should first explain scene parsing in  a line or so...what it does is to give global information context...see https://towardsdatascience.com/review-pspnet-winner-in-ilsvrc-2016-semantic-segmentation-scene-parsing-e089e5df177d</p><p> </p><p>For example: Semantic segmentation is used to classify each pixel of an image for only labelled objects. However, scene parsing, which is based on semantic segmentation has the capability to classify all pixels within an image. It takes into account the global context of the image to predict the local level predictions hence gives better performance on benchmark datasets like PASCAL VOC 2012 and cityscapes.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/547/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Reply:
<p>I think we don't need to tell the subtle differences between scene-parsing and semantic segmentation. Therefore, I will not mention much about scene parsing.</p><p> </p><p>Rest changes I have incorporated from this part.</p>

Reply:
<blockquote>In the guide [How u-net works](https://developers.arcgis.com/python/guide/how-unet-works/), we have learned in detail about semantic segmentation. There are many semantic segmentation algorithms such as U-net, PSPNet, Feature Pyramid Network (FPN), etc. In this guide, we will mainly focus on Pyramid scene parsing network (PSPNet) [1] which is one of the most well-recognized image segmentation algorithms and many of the ideas are shared among other algorithms.</blockquote><blockquote>Note: To follow the guide below, we assume that you have some basic understanding of the convolutional neural networks (CNN) concept. You can refresh your CNN knowledge by going through this short paper “[A guide to convolution arithmetic for deep learning](https://arxiv.org/pdf/1603.07285.pdf)".</blockquote><p> </p><p>In the guide [How u-net works](https://developers.arcgis.com/python/guide/how-unet-works/), we have learned in detail about semantic segmentation. There are many semantic segmentation algorithms such as U-net, PSPNet, Feature Pyramid Network (FPN), etc. In this guide, we will mainly focus on Pyramid scene parsing network (PSPNet) [1] which is one of the most well-recognized image segmentation algorithms and many of the ideas are shared among other algorithms.</p><p>Note: To follow the guide below, we assume that you have conceptual understanding of "Convolutional Neural Networks (CNN)". You can refresh your CNN knowledge by going through this short paper “[A guide to convolution arithmetic for deep learning](https://arxiv.org/pdf/1603.07285.pdf)".</p>